### DISCUSSION TITLE
How can I update map fileds.

### DISCUSSION THREAD

**Discussion by v1xingyue** [antonmedv/expr#588] [General]
I have some map fields in a Env , and a string content . Env struct is like this :

```golang
type Env struct {
    msg string
   fileds map[string]any
}
```

Now I want to add filed by the msg content , such as . 

```golang
fileds["len"] = len(msgs)
fileds["uppper"] = uppper(msgs)
```

At last return the fields to the expr script caller. 

How can I do that. 

Thanks. 

And I wonder why not provide built in much more map functions.

**Comment by antonmedv** [comment]
I didn’t get the question.  Expr doesn’t support assignment operations. You can add them as functions.
